examples/scaleway/main.tf

```diff
@@ -1,54 +1,42 @@
-locals {
-  cluster_name   = "scaleway-test"
-  cluster_region = "fr-par"
-  cluster_zone   = "fr-par-1"
-  tags           = ["test", "${local.cluster_name}"]
-}
-
-module "cluster" {
-  source = "git::https://github.com/camptocamp/devops-stack.git//modules/scaleway?ref=v1-alpha"
-
-  kubernetes_version = "1.24.3"
-
-  cluster_type = "kapsule"
-  cluster_name = local.cluster_name
-  cluster_tags = local.tags
-  region       = local.cluster_region
-  zone         = local.cluster_zone
-  lb_type      = "LB-S"
+# ###########################
+# INFRA + K8s PHASE
+# ###########################
+module "scaleway" {
+  source = "git@github.com:camptocamp/devops-stack-module-cluster-scaleway.git"
 
+  base_domain         = var.base_domain
+  cluster_name        = var.cluster_name
+  cluster_description = var.cluster_description
+  cluster_tags        = var.cluster_tags
+  cluster_type        = var.cluster_type
+  kubernetes_version  = var.kubernetes_version
+  lb_name             = var.lb_name
+  lb_type             = var.lb_type
+  zone                = var.zone
+  node_pools          = var.node_pools
 }
 
+# ###########################
+# BOOTSPRAP APPLICATION PHASE
+# ###########################
 
 module "argocd_bootstrap" {
-  source         = "git::https://github.com/camptocamp/devops-stack-module-argocd.git//bootstrap?ref=v1-alpha"
-  cluster_name   = local.cluster_name
-  base_domain    = module.cluster.base_domain
-  cluster_issuer = "letsencrypt-prod"
-
-  argocd = {
-    admin_enabled = "true"
-  }
-
-  depends_on = [
-    module.cluster,
-  ]
+  source = "git::https://github.com/camptocamp/devops-stack-module-argocd.git//bootstrap?ref=v4.4.0"
 }
 
+module "ingress_controller" {
+  source = "git::https://github.com/camptocamp/devops-stack-module-traefik.git?ref=v5.0.0"
 
-module "ingress" {
-  source = "git::https://github.com/camptocamp/devops-stack-module-traefik.git//scaleway?ref=v1-alpha"
-
-  cluster_name     = local.cluster_name
-  argocd_namespace = module.argocd_bootstrap.argocd_namespace
-  base_domain      = module.cluster.base_domain
+  cluster_name           = var.cluster_name
+  base_domain            = module.scaleway.base_domain
+  enable_service_monitor = var.ingress_enable_service_monitor
 
   helm_values = [{
     traefik = {
       service = {
         type = "LoadBalancer"
         annotations = {
-          "service.beta.kubernetes.io/scw-loadbalancer-id" = module.cluster.lb_id
+          "service.beta.kubernetes.io/scw-loadbalancer-id" = module.scaleway.lb_id
         }
       }
     }
```

Comment:
If this a required setting for the Traefik module, maybe a new variant of that module should be created. I see that this is very similar to what was done on the SKS module.

Using the `helm_values` variable should be discouraged and used only for edge cases.